src/layouts/PartnersSection.jsx

```diff
@@ -0,0 +1,204 @@
+import PropTypes from 'prop-types'
+
+import Kite from '@/assets/images/icn-kite.png'
+import CTAButton from '@/components/ui/CTAButton'
+
+const PartnersSection = ({
+  partnersData = {},
+  year = new Date().getFullYear(),
+}) => {
+  const isCurrentYear = year === new Date().getFullYear()
+
+  // Check if there's any partner data
+  const hasPartners =
+    (partnersData.platinum && partnersData.platinum.length > 0) ||
+    (partnersData.diamond && partnersData.diamond.length > 0) ||
+    (partnersData.gold && partnersData.gold.length > 0) ||
+    (partnersData.organizations && partnersData.organizations.length > 0)
+
+  const renderPartnerTier = (
+    tierName,
+    partners,
+    isSideBySide = false,
+    useFourColumns = false,
+    useSingleColumn = false
+  ) => {
+    if (!partners || partners.length === 0) return null
+
+    return (
+      <div className="mb-12">
+        <h3 className="mb-6 py-4 text-center font-russell text-5xl text-black">
+          {tierName}
+        </h3>
+        <div
+          className={`grid gap-8 ${
+            useSingleColumn
+              ? 'grid-cols-1 justify-items-center'
+              : useFourColumns
+                ? 'grid-cols-1 justify-items-stretch sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4'
+                : isSideBySide
+                  ? 'grid-cols-1 justify-items-center sm:grid-cols-2'
+                  : 'grid-cols-1 justify-items-center sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4'
+          }`}
+        >
+          {partners.map((partner) => (
+            <button
+              key={partner.id}
+              className="group relative flex justify-center p-4 transition-transform duration-200 hover:scale-105"
+              onClick={() => window.open(partner.url, '_blank')}
+            >
+              {partner.logo ? (
+                <img
+                  src={partner.logo}
+                  alt={partner.name}
+                  className="h-24 w-full max-w-72 object-contain transition-opacity duration-200 group-hover:opacity-90"
+                />
+              ) : (
+                <div className="flex h-24 items-center justify-center rounded-lg border-2 border-gray-300 bg-gray-50 px-6">
+                  <p className="text-center text-lg font-semibold text-gray-700">
+                    {partner.name}
+                  </p>
+                </div>
+              )}
+              <div className="invisible absolute left-0 top-0 flex size-full flex-col items-center justify-center overflow-y-hidden rounded bg-black/50 p-4 text-white backdrop-blur-sm transition-all duration-200 group-hover:visible">
+                <h4 className="mb-2 text-xl font-semibold md:text-2xl">
+                  {partner.name}
+                </h4>
+                <p className="overflow-y-auto text-left text-sm md:text-base">
+                  {partner.desc}
+                </p>
+              </div>
```

Comment:
I just tested this and it doesn't work that well for the WSU logo or the WSU GDG due to the partner name covering the entire area and no partner description available for overflow scrolling. I think this can be improved in the future